### PR TITLE
Tk 112 refactor timer

### PR DESCRIPTION
**Pull request by ikbennadine**
[TK-112 The Timer class should not reset itself on complete.](https://codedazur.atlassian.net/browse/TK-112)
[TK-113 A completed timer should not be resumable.](https://codedazur.atlassian.net/browse/TK-113)

### CODE REVIEW

File: packages/essentials/utilities/timing/Timer.test.ts [hunk 1]
```diff
@@ -149,6 +159,145 @@ describe("Timer", () => {
...
+    it("should update the timer duration and update the remaining time when paused", () => {
+      timer.start();
+      timer.pause();
+      timer.setDuration(7000);
+
+      expect(timer.duration).toBe(7000);
+      expect(timer.remaining).toBe(7000);
+    });
+
+    it("should extend a completed timer duration and show correct status", () => {
```
Comment (sayinserdar) [inline packages/essentials/utilities/timing/Timer.test.ts:188]: Nitpicking but do we need to run linters for this file? 😅

File: packages/essentials/utilities/timing/Timer.ts [hunk 2]
```diff
@@ -45,11 +51,15 @@ export class Timer {
   }
 
   public get status(): TimerStatus {
```
Comment (sayinserdar) [inline packages/essentials/utilities/timing/Timer.ts:53]: clean!

**Review by sayinserdar — APPROVED** [review]

File: packages/essentials/utilities/timing/Timer.test.ts (hunk 1 above)
Comment (ikbennadine) [reply packages/essentials/utilities/timing/Timer.test.ts:188]: Already did, this is just looking weird because github is breaking it to the next line.

**Review by thijsdaniels — APPROVED** [review]
looks great to me dien!

File: packages/essentials/utilities/timing/Timer.test.ts [hunk 3]
```diff
@@ -149,6 +159,145 @@ describe("Timer", () => {
...
+    it("should not complete a paused timer when duration is set to less then the remaining time", () => {
+      timer.start();
+      vi.advanceTimersByTime(2000);
+      timer.pause();
+
+      expect(timer.status).toBe(TimerStatus.paused);
+
+      timer.setDuration(2000);
+
+      expect(timer.status).toBe(TimerStatus.paused);
```
Comment (ikbennadine) [inline packages/essentials/utilities/timing/Timer.test.ts:283]:
```suggestion
      expect(timer.status).toBe(TimerStatus.paused);
      expect(callback).not.toHaveBeenCalled();
```

**Pull request merged**